modules/module_loader/include/ModulesApp.h

```diff
@@ -21,7 +100,131 @@ class ModulesApp : public MooseApp
 
   static void registerApps();
   static void registerAll(Factory & f, ActionFactory & af, Syntax & s);
+  template <typename T>
+  static void registerAllObjects(Factory & f, ActionFactory & af, Syntax & s);
   static void registerObjects(Factory & factory);
   static void associateSyntax(Syntax & syntax, ActionFactory & action_factory);
   static void registerExecFlags(Factory & factory);
 };
+
+///@{
+/**
+ * Dummy methods to clear unused parameter warnings from compiler
+ */
+void
+clearUnusedWarnings(Factory & /*factory*/)
+{
+}
+void
+clearUnusedWarnings(Syntax & /*syntax*/, ActionFactory & /*action_factory*/)
+{
+}
+///@}
+
+template <typename T>
+void
+ModulesApp::registerAllObjects(Factory & f, ActionFactory & af, Syntax & s)
+{
+#ifdef CHEMICAL_REACTIONS_ENABLED
+  ChemicalReactionsApp::registerAll(f, af, s);
+#endif
+
+#ifdef CONTACT_ENABLED
+  ContactApp::registerAll(f, af, s);
+#endif
+
+#ifdef ELECTROMAGNETICS_ENABLED
+  ElectromagneticsApp::registerAll(f, af, s);
+#endif
+
+#ifdef FLUID_PROPERTIES_ENABLED
+  FluidPropertiesApp::registerAll(f, af, s);
+#endif
+
+#ifdef FSI_ENABLED
+  FsiApp::registerAll(f, af, s);
+#endif
+
+#ifdef GEOCHEMISTRY_ENABLED
+  GeochemistryApp::registerAll(f, af, s);
+#endif
+
+#ifdef HEAT_CONDUCTION_ENABLED
+  HeatConductionApp::registerAll(f, af, s);
+#endif
+
+#ifdef LEVEL_SET_ENABLED
+  LevelSetApp::registerAll(f, af, s);
+#endif
+
+#ifdef MISC_ENABLED
+  MiscApp::registerAll(f, af, s);
+#endif
+
+#ifdef NAVIER_STOKES_ENABLED
+  NavierStokesApp::registerAll(f, af, s);
+#endif
+
+#ifdef OPTIMIZATION_ENABLED
+  OptimizationApp::registerAll(f, af, s);
+#endif
+
+#ifdef PERIDYNAMICS_ENABLED
+  PeridynamicsApp::registerAll(f, af, s);
+#endif
+
+#ifdef PHASE_FIELD_ENABLED
+  PhaseFieldApp::registerAll(f, af, s);
+#endif
+
+#ifdef POROUS_FLOW_ENABLED
+  PorousFlowApp::registerAll(f, af, s);
+#endif
+
+#ifdef RAY_TRACING_ENABLED
+  RayTracingApp::registerAll(f, af, s);
+#endif
+
+#ifdef RDG_ENABLED
+  RdgApp::registerAll(f, af, s);
+#endif
+
+#ifdef REACTOR_ENABLED
+  ReactorApp::registerAll(f, af, s);
+#endif
+
+#ifdef RICHARDS_ENABLED
+  RichardsApp::registerAll(f, af, s);
+#endif
+
+#ifdef SCALAR_TRANSPORT_ENABLED
+  ScalarTransportApp::registerAll(f, af, s);
+#endif
+
+#ifdef SOLID_PROPERTIES_ENABLED
+  SolidPropertiesApp::registerAll(f, af, s);
+#endif
+
+#ifdef STOCHASTIC_TOOLS_ENABLED
+  StochasticToolsApp::registerAll(f, af, s);
+#endif
+
+#ifdef TENSOR_MECHANICS_ENABLED
+  TensorMechanicsApp::registerAll(f, af, s);
+#endif
+
+#ifdef THERMAL_HYDRAULICS_ENABLED
+  ThermalHydraulicsApp::registerAll(f, af, s);
+#endif
+
+#ifdef XFEM_ENABLED
+  XFEMApp::registerAll(f, af, s);
+#endif
+
+#ifdef EXTERNAL_PETSC_SOLVER_ENABLED
+  ExternalPetscSolverApp::registerAll(f, af, s);
+#endif
+
+  clearUnusedWarnings(f);
+  clearUnusedWarnings(s, af);
```

Comment:
```suggestion
  libmesh_ignore(f, s, af);
```

Reply:
Any thoughts on why these are here in the first place? It was a I don't understand what these are for so don't touch them, "inaction".

Reply:
If no modules are defined, then these variables would be unused